src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -89,6 +91,8 @@ public function getUser()
         if (null === $this->tokenStorage) {
             if (null === $this->container) {
                 throw new \RuntimeException('The "app.user" variable is not available.');
+            } elseif (!$this->container->has('security.context')) {
+                return;
```

Comment:
an exception saying it is not available (as done in the previous case) may make more sense